File: scratch/test_gemini.py
import urllib.request
import json
import base64

KEY_B64 = 'QVEuQWI4Uk42S01UdnoxZnQ3al9TRmpFaVB6dnJwQVhreC1PU3hOU2ZyczByd1E1SVZBUFE='
gemini_key = base64.b64decode(KEY_B64).decode('utf-8')
print("Decoded Key:", gemini_key[:10] + "...")

models = ['gemini-2.0-flash', 'gemini-1.5-flash', 'gemini-2.0-flash-exp']
for m in models:
    url = f"https://generativelanguage.googleapis.com/v1beta/models/{m}:generateContent?key={gemini_key}"
    req = urllib.request.Request(
        url,
        data=json.dumps({
            "contents": [{"parts": [{"text": "Reply JSON: {\"status\": \"ok\"}"}]}],
            "generationConfig": {"responseMimeType": "application/json"}
        }).encode('utf-8'),
        headers={"Content-Type": "application/json"},
        method="POST"
    )
    try:
        with urllib.request.urlopen(req) as response:
            res_text = response.read().decode('utf-8')
            print(f"Model {m} SUCCESS (200):", res_text[:150])
    except Exception as e:
        print(f"Model {m} ERROR:", e)

File: scratch/test_gemini.js
const KEY_B64 = 'QVEuQWI4Uk42S01UdnoxZnQ3al9TRmpFaVB6dnJwQVhreC1PU3hOU2ZyczByd1E1SVZBUFE=';
const apiKey = process.env.GEMINI_API_KEY || Buffer.from(KEY_B64, 'base64').toString('utf-8');

async function testModels() {
  const models = ['gemini-2.5-flash', 'gemini-2.0-flash', 'gemini-1.5-flash', 'gemini-flash-latest', 'gemini-3.6-flash'];
  for (const m of models) {
    try {
      const endpoint = `https://generativelanguage.googleapis.com/v1beta/models/${m}:generateContent?key=${apiKey}`;
      const res = await fetch(endpoint, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({
          contents: [{ parts: [{ text: "Respond with json: {\"status\": \"ok\"}" }] }],
          generationConfig: { responseMimeType: "application/json" }
        })
      });
      console.log(`Model ${m}: HTTP ${res.status}`);
      if (res.ok) {
        const d = await res.json();
        console.log(`Result ${m}:`, d.candidates?.[0]?.content?.parts?.[0]?.text);
      }
    } catch (e) {
      console.log(`Model ${m} error:`, e.message);
    }
  }
}

testModels();
